src/pck/mod.rs

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+
+mod raw;
+
+use raw::SgxExtensionRaw;
+
+use const_oid::ObjectIdentifier;
+use der::Decode;
+use x509::ext::Extensions;
+
+const SGX_EXTENSION: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.113741.1.13.1");
+
+pub struct SgxExtension<'a> {
+    pub fmspc: &'a [u8],
+    pub pcesvn: u8,
+    pub pceid: &'a [u8],
+    pub tcb_components: [u8; 16],
+    pub is_multi: bool,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum SgxExtensionError {
+    NoExtensionsFound,
+    MissingSgxExtension,
+    DerDecodingError(der::Error),
+}
+
+impl core::fmt::Display for SgxExtensionError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            SgxExtensionError::NoExtensionsFound => write!(f, "SGX: No extensions found"),
+            SgxExtensionError::MissingSgxExtension => write!(f, "SGX: Missing extension"),
+            SgxExtensionError::DerDecodingError(e) => write!(f, "SGX: Der decoding error: {e}"),
+        }
+    }
+}
+
+impl<'a> SgxExtension<'a> {
+    pub fn from_x509_extensions(extensions: &'a Extensions) -> Result<Self, SgxExtensionError> {
+        if extensions.is_empty() {
+            return Err(SgxExtensionError::NoExtensionsFound);
+        }
+
+        for extension in extensions.iter() {
```

Comment:
is `extensions.iter().find(...)` better?

Reply:
Wow, that's a decent amount of code reduction. I didn't know about `.iter().find()`